src/windows/models/properties_model.py

```diff
@@ -542,6 +571,228 @@ def value_updated(self, item, interpolation=-1, value=None, interpolation_detail
             # Clear selection
             self.parent.clearSelection()
 
+    def add_property(self, property, filter, c, item_type, object_id=None):
```

Comment:
Kind of a nitpick, but I'd rather name this something like `set_property` or `update_property`, since it _both_ creates and modifies property values. (`add_property()` would be the bit of the code just inside the `if self.new_item:` condition.)